src/session.ts

```diff
@@ -22,36 +30,117 @@ export type State =
   | { connection: Connection.Connecting }
   | { connection: Connection.Connected; session: Session };
 
+export type ConnectionModalState =
+| { status: ConnectionModalStateType.Open; modalProps: ConnectionModalProps }
+| { status: ConnectionModalStateType.Close; modalProps: null };
 export interface Session {
   address: string;
   tokenBalance: BigNumber | null; // `null` means it isn't loaded yet.
   tx: TxState;
 }
-
+export interface ConnectionModalProps {
+  config: Config;
+  uri: string;
+}
 export interface Store extends Readable<State> {
-  connect(config: Config): Promise<void>;
+  connectMetamask(config: Config): Promise<void>;
   updateBalance(n: BigNumber): void;
   refreshBalance(config: Config): Promise<void>;
-
+  provider: ethers.providers.Provider;
+  signer: ethers.Signer;
+  connectWalletConnect(config: Config): Promise<void>;
   setTxSigning(): void;
   setTxPending(tx: TransactionResponse): void;
   setTxConfirmed(tx: TransactionReceipt): void;
   setChangedAccount(address: string): void;
 }
+let walletConnect: any;
+
+const modalStore = writable<ConnectionModalState>({ status: ConnectionModalStateType.Close, modalProps: null });
+export const store = derived(modalStore, ($store) => $store);
 
 export const loadState = (initial: State): Store => {
   const store = writable<State>(initial);
   const session = window.localStorage.getItem("session");
-
+  function newWalletConnect(config: Config): WalletConnect {
+
+    walletConnect = new WalletConnect({
+      bridge: config.radicleBridge.bridge,
+      qrcodeModal: {
+        open: (uri: string, onClose, _opts?: unknown) => {
+          modalStore.set({ status: ConnectionModalStateType.Open, modalProps: { uri, config } });
+          console.log(_opts);
+          onClose();
+        },
+        close: () => {
+          modalStore.set({ status: ConnectionModalStateType.Close, modalProps: null });
+        },
+      },
+    });
```

Comment:
This instantiation of wallet connect includes settings for the qrcodeModal, while the other in Config doesn't, why is that?

Reply:
The one in the config loads when the app renders or reloads and if it has the qrcode modal, it'll popup the qrcode modal every single time

Reply:
anyways with the global walletconnect that's now a attribute of the class, I don't need the wallet connect instantiation in the config.ts